tools/cassandra/version.go

```diff
@@ -94,3 +98,66 @@ func parseValidateVersion(ver string) (string, error) {
 	}
 	return ver, nil
 }
+
+// versionManifest is a value type that represents the deserialized version.json file within
+// a schema directory
+type versionManifest struct {
+	ExpectedVersion string
+}
+
+// getExpectedVersion gets the expected version
+func getExpectedVersion(dirPath string) (string, error) {
+	filePath := dirPath + "/version.json"
+	jsonStr, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	jsonBlob := []byte(jsonStr)
+	var manifest versionManifest
+	err = json.Unmarshal(jsonBlob, &manifest)
+	if err != nil {
+		return "", err
+	}
+
+	currVer, err := parseValidateVersion(manifest.ExpectedVersion)
+	if err != nil {
+		return "", fmt.Errorf("invalid ExpectedVersion in manifest: %s", err.Error())
+	}
+	return currVer, nil
+}
+
+func CheckCompatibleVersion(cfg config.Cassandra, keyspace string, dirPath string) error {
+	cqlClient, err := NewCQLClient(cfg.Hosts, cfg.Port, cfg.User, cfg.Password, keyspace)
+	if err != nil {
+		return errors.New("Unable to create CQL Client: " + err.Error())
+	}
+	defer cqlClient.Close()
+	version, err := cqlClient.ReadSchemaVersion()
+	if err != nil {
+		return errors.New("Unable to create schema version: " + err.Error())
+	}
+	expectedVersion, err := getExpectedVersion(dirPath)
+	if err != nil {
+		return errors.New("Unable to read expected schema version: " + err.Error())
+	}
+	expectedNum, err := strconv.ParseFloat(expectedVersion, 64)
+	if err != nil {
+		return errors.New("Unable to convert expected version to float: " + expectedVersion)
+	}
+	actualNum, err := strconv.ParseFloat(version, 64)
+	if err != nil {
+		return errors.New("Unable to convert actual version to float: " + version)
+	}
+	// In most cases, the versions should match. However if after a schema upgrade there is a code
+	// rollback, the code version (expected version) would fall lower than the actual version in
+	// cassandra. This check is to allow such rollbacks since we only make backwards compatible schema
+	// changes
+	if actualNum <= expectedNum {
```

Comment:
i think this check should be actual < expected. also fix the unit test

Reply:
also, avoid comparing floats directly when possible. This file has a cmpVersion(a, b) method at the top - so you can directly compare two version strings.